guides/git-workflow.md

```diff
@@ -1,21 +1,89 @@
+# Table of contents
+1. [Introduction](#introduction)
+2. [Which workflow should I use?](#which-workflow)
+    1. [Branching Workflow](#branching-pros-cons)
+    2. [Forking Workflow](#forking-pros-cons)
+3. [Workflow Instructions](#workflow-instructions)
+    1. [Opening a GitHub Issue](#github-issue)
+    2. [Branching Workflow](#branching-workflow)
+    3. [Forking Workflow](#forking-workflow)
+        1. [Keeping your fork updated](#update-fork)
+        2. [Collaborating on someone else's fork](#collaborate-on-fork)
 
-### Opening a GitHub Issue
-Before contributing, you should determine if the eventual change is significant enough to warrant a GitHub issue; if you think that this change will be solving a significant problem or adding a providing a significant enhancement to the project, then it would be advantageous to open an issue under the main GitHub repository. This will allow both contributors and repository maintainers to keep track of the project and capture any needed context of the particular change. Any appropriate individuals should be assigned to the issue, and any appropriate label(s) should be tagged.
 
-### Workflow
-Any changes that you want to eventually make to the `master` branch of the `spacetelescope` repository should be done through a workflow where you [create a fork](https://help.github.com/articles/fork-a-repo/) and work on a personal branch before submitting those changes to be reviewed through a pull request. Instructions on how to do those things can be found below:
+## Introduction <a name="introduction"></a>
+The following describes two different workflows for contributing software to an open-source repository on GitHub under the `spacetelescope` organization (hereby referred to as the "`spacetelescope` repository"). One model involves creating branches off of the `spacetelescope` repository directly (hereby referred to as the "branching workflow"), while the other involves forking the `spacetelescope` repository and developing changes on personal branches of the fork (hereby referred to as the "forking workflow").  Regardless of which model is chosen, software changes ultimately get incorporated into the `spacetelescope` repository through pull requests.  Below, we provide some pros/cons for each workflow.  We encourage developers to chose a workflow that is most appropriate for your repository.
+
+## Which workflow should I use? <a name="which-workflow"></a>
+There are various pros and cons to each workflow, and it may not be obvious as to which workflow is best to choose for your project.  However, the power and benefit of the workflow comes in its consistent use across all team members and collaborators during the development of the software.
+
+### Branching Workflow <a name="branching-pros-cons"></a>
+
+**Pros:**
```

Comment:
Another substantial pro for this workflow is: 
- Easily allows multiple developers to contribute commits to a single feature or PR. Any team member with write access to the repository can collaborate directly in adding code, revising, or testing that branch.

Reply:
Multiple people working directly on the same branch can easily become a "con" if there is insufficient coordination. 🤷

Reply:
right, but it's also literally required in other cases! That's my whole point, one team's pro is another team's con, so there's not one right way of doing this all. :-)

Reply:
To Marshall's "one team's pro is another team's con" point -- I am wondering if we should not have a pros/cons list at all, but rather a single list of "things to consider".  Calling something a "pro" or a "con" can tend to carry a lot of weight 🤔